### DISCUSSION TITLE
MakeRotLib for NCAA, modifying parameter file

### DISCUSSION THREAD

**Discussion by wendi-luo** [RosettaCommons/rosetta#252] [Rosetta] [answered]
I'm making rotlib for NCAA, I already made rotlibs file, but I'm confused about "Modifying the ResidueType Parameter File" step. 
![image](https://github.com/user-attachments/assets/0382dbd2-91b2-4dfa-8e2a-c1d9bf9d3a99)

1. echo "NCAA_ROTLIB_PATH C40.rotlib" >> PATH/TO/rosetta/main/database/chemical/residue_type_sets/fa_standard/residue_types/l-ncaa/ornithine.params    
"NCAA_ROTLIB_PATH " should be "PATH/TO/rosetta/main/database/ncaa_rotlibs/" or just type "NCAA_ROTLIB_PATH"?
2. echo "NCAA_ROTLIB_NUM_ROTAMER_BINS 3 3 3 3" >> PATH/TO/rosetta/main/database/chemical/residue_type_sets/fa_standard/residue_types/l-ncaa/ornithine.params  
'3 3 3 3' what does each of the 3 stand for??  cuz my NCAA just have 1 dihedral, Idk how to write this line.

Thank u for your response.

**Comment by roccomoretti** [comment, marked as answer]
NCAA_ROTLIB_PATH is the literal string "NCAA_ROTLIB_PATH"

For the NCAA_ROTLIB_NUM_ROTAMER_BINS line, the first entry is the number of rotameric chis. The subsequent entries are the number of bins for each rotameric chi. So for a single rotameric dihedral, you'd have `NCAA_ROTLIB_NUM_ROTAMER_BINS 1 3` (or `NCAA_ROTLIB_NUM_ROTAMER_BINS 1 2`, etc., depending on how many bins you have for your one dihedral.)
